bus/EasyCaching.Bus.RabbitMQStream/DefaultRabbitMQStreamBus.cs

```diff
@@ -0,0 +1,232 @@
+    using EasyCaching.Core;
+    using EasyCaching.Core.Bus;
+    using EasyCaching.Core.Serialization;
+    using global::RabbitMQ.Client;
+    using global::RabbitMQ.Client.Events;
+    using Microsoft.Extensions.ObjectPool;
+    using Microsoft.Extensions.Options;
+
+    /// <summary>
+    /// Default RabbitMQ Bus.
+    /// </summary>
+    public class DefaultRabbitMQStreamBus : EasyCachingAbstractBus
+    {
+        /// <summary>
+        /// The subscriber connection.
+        /// </summary>
+        private readonly IConnection _subConnection;
+
+        /// <summary>
+        /// The publish channel pool.
+        /// </summary>
+        private readonly ObjectPool<IModel> _pubChannelPool;
+
+        /// <summary>
+        /// The rabbitMQ Bus options.
+        /// </summary>
+        private readonly RabbitMQBusOptions _options;
+
+        /// <summary>
+        /// The serializer.
+        /// </summary>
+        private readonly IEasyCachingSerializer _serializer;
+
+        /// <summary>
+        /// The identifier.
+        /// </summary>
+        private readonly string _busId;
+
+        private static readonly Dictionary<string, object> _streamArgs = new Dictionary<string, object>
+        {
+            { "x-queue-type", "stream" },
+            { "x-max-age", "5m" },
+            { "x-stream-max-segment-size-bytes", 4_000_000 }
+        };
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="T:EasyCaching.Bus.RabbitMQ.DefaultRabbitMQBus"/> class.
+        /// </summary>
+        /// <param name="_objectPolicy">Object policy.</param>
+        /// <param name="rabbitMQOptions">RabbitMQ Options.</param>
+        /// <param name="serializer">Serializer.</param>
+        public DefaultRabbitMQStreamBus(
+            IPooledObjectPolicy<IModel> _objectPolicy
+            , IOptions<RabbitMQBusOptions> rabbitMQOptions
+            , IEasyCachingSerializer serializer)
+        {
+            this._options = rabbitMQOptions.Value;
+            this._serializer = serializer;
+
+            var factory = new ConnectionFactory
+            {
+                HostName = _options.HostName,
+                UserName = _options.UserName,
+                Port = _options.Port,
+                Password = _options.Password,
+                VirtualHost = _options.VirtualHost,
+                RequestedConnectionTimeout = System.TimeSpan.FromMilliseconds(_options.RequestedConnectionTimeout),
+                SocketReadTimeout = System.TimeSpan.FromMilliseconds(_options.SocketReadTimeout),
+                SocketWriteTimeout = System.TimeSpan.FromMilliseconds(_options.SocketWriteTimeout),
+                ClientProvidedName = _options.ClientProvidedName
+            };
+
+            _subConnection = factory.CreateConnection();
+
+            _subConnection.ConnectionShutdown += (_, e) =>
+            {
+                _options._logger?.Invoke($"ERROR: ConnectionShutdown: {e.ReplyText}");
```

Comment:
Using `ILogger` directly may be better here.

Reply:
If I remember correctly it's not trivial to use Dependency Injection incident this package, do you suggest injecting ILogger instead of the callback (Action<string>) on the RabbitMQBusOptions?

Reply:
As an alternative I can modify .WithRabbitMQStreamBus(options) to accept the ILogger as a second argument in order to remove any modifications to the RabbitMQBusOptions.cs